lua/core/menu.lua

```diff
@@ -1145,7 +1146,104 @@ m.deinit[pRESET] = function() end
 
 
 -----------------------------------------
+-- UPDATE
+m.update = {}
+m.update.url = ''
+m.update.version = ''
+
+local function check_newest()
+  print("checking for update")
+  m.update.url = util.os_capture( [[curl -s \
+      https://api.github.com/repos/monome/norns/releases/latest \
+      | grep "browser_download_url.*" \
+      | cut -d : -f 2,3 \
+      | tr -d \"]])
+  print(m.update.url)
+  m.update.version = m.update.url:match("(%d%d%d%d%d%d)")
+  print("available version "..m.update.version)
+end
+
+local function get_update()
+  m.update.message = "preparing..."
+  menu.redraw()
+  pcall(cleanup) -- shut down script
+  norns.script.clear()
+  os.execute("sudo systemctl stop norns-jack.service") -- disable audio
```

Comment:
should this all be wrapped in an exception handler somehow?  is there any realistic chance of anything failing?

Reply:
ok, sorry.  looks like best we could do is:

```lua
local code = os.execute(...)
```

maybe not worth it. 🤷‍♀️

Reply:
the `pcall` failing is no big deal. it just won't issue a warning.